src/tz.cpp

```diff
@@ -267,6 +294,83 @@ get_download_folder()
 
 #  endif  // !_WIN32
 
+
+/*
+ * This class is provided to mimic the following usage of `ifstream`:
+ *
+ * std::ifstream is(filename);
+ *
+ * file_streambuf ibuf(filename);
+ * std::istream is(&ibuf);
+ *
+ * This is required because `ifstream` does not support opening files
+ * containing wide characters on Windows. On Windows, `file_streambuf` uses
+ * `file_open()` to convert the file name to UTF-16 before opening it with
+ * `_wfopen()`.
+ *
+ * Note that this is not an exact re-implementation of `ifstream`,
+ * but is enough for our usage.
+ *
+ * It is partially based on these two implementations:
+ * - fdinbuf from http://www.josuttis.com/cppcode/fdstream.html
+ * - stdiobuf https://stackoverflow.com/questions/12342542/convert-file-to-ifstream-c-android-ndk
+ *
+ * Apparently MSVC provides non-standard overloads of `ifstream` that support
+ * a `const wchar_t*` file name, but MinGW does not.
+ * https://stackoverflow.com/a/822032
+ */
+class file_streambuf
+  : public std::streambuf
+{
+private:
+    FILE* file_;
+    static const int buffer_size_ = 1024;
+    char buffer_[buffer_size_];
+
+public:
+    file_streambuf(const std::string& filename)
+        : file_(file_open(filename))
+    {
+    }
+
+    ~file_streambuf()
+    {
+        if (file_) {
+            ::fclose(file_);
+        }
+    }
+
+protected:
+    virtual int_type underflow() {
```

Comment:
This seems to be the only method we need to override from `std::streambuf` for this to work? I learned this from SO and the fdinbuf implementations, and it seems to work fine for any usage of the `istream` here.